File: src/main/java/com/example/citycontrol/service/CityService.java
package com.example.citycontrol.service;

import com.example.citycontrol.entity.City;
import com.example.citycontrol.filter.CityFilterData;
import java.io.File;
import java.util.Collection;

public interface CityService
{
	boolean addCity(City city);

	boolean removeCity(Long id);

	boolean updateCity(City city);

	City getCity(Long id);

	Collection<City> getAllCities();

	Collection<City> getCitiesByFilter(CityFilterData filterData);

	void importCsv(File file);

	Collection<Object[]> countCitiesByGroup(String propName);
}

File: client/src/services/CityService.js
import http from "../http-common";

class CityService {
    constructor() {
        let self = this;
        self.cityRestUrl = "/cities";
    }

    addCity(city) {
        return http.post(this.cityRestUrl, city);
    }

    updateCity(city) {
        return http.put(`${this.cityRestUrl}`, city);
    }

    removeCity(id) {
        return http.delete(`${this.cityRestUrl}/${id}`);
    }

    getAllCities() {
        return http.get(this.cityRestUrl);
    }

    getCitiesByFilter(filterParams) {
        return http.put(`${this.cityRestUrl}/filter`, filterParams);
    }

    getCity(id) {
        return http.get(`${this.cityRestUrl}/${id}`);
    }
}

export default new CityService();